OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs

```diff
@@ -132,8 +133,9 @@ public static IEnumerable<string> LinterHotkeyNames(MiniYamlNode widgetNode, Act
 		}
 
 		[ObjectCreator.UseCtor]
-		public ViewportControllerWidget(World world, WorldRenderer worldRenderer)
+		public ViewportControllerWidget(ModData modData, World world, WorldRenderer worldRenderer)
```

Comment:
Isn't it a bit overkill to pass the entire `ModData` here (as well as 20 other places) for the `HotkeyManager`?

Reply:
...no?

We support dynamic arguments for the logic arguments, world, world renderer, and order manager.  Adding arguments for all the various state that these top-level types own IMO pushes us down a slippery slope of people wanting us to support more and more unnecessary and possibly mod-specific state.